framework/src/systems/NonlinearSystem.C

```diff
@@ -234,29 +212,27 @@ NonlinearSystem::solve()
 }
 
 void
-NonlinearSystem::stopSolve()
+NonlinearSystem::stopSolve(const ExecFlagType & exec_flag)
 {
   PetscNonlinearSolver<Real> & solver =
       static_cast<PetscNonlinearSolver<Real> &>(*sys().nonlinear_solver);
-  SNESSetFunctionDomainError(solver.snes());
-
-  // Insert a NaN into the residual vector.  As of PETSc-3.6, this
-  // should make PETSc return DIVERGED_NANORINF the next time it does
-  // a reduction.  We'll write to the first local dof on every
-  // processor that has any dofs.
-  _nl_implicit_sys.rhs->close();
-
-  if (_nl_implicit_sys.rhs->local_size())
-    _nl_implicit_sys.rhs->set(_nl_implicit_sys.rhs->first_local_index(),
-                              std::numeric_limits<Real>::quiet_NaN());
-  _nl_implicit_sys.rhs->close();
-
-  // Clean up by getting other vectors into a valid state for a
-  // (possible) subsequent solve.  There may be more than just
-  // these...
-  if (_Re_time)
-    _Re_time->close();
-  _Re_non_time->close();
+
+  if (exec_flag == EXEC_LINEAR || exec_flag == EXEC_POSTCHECK)
```

Comment:
I assume these should match what is in `FEProblemBase::checkExceptionAndStopSolve()`, right? Is it intentional that `EXEC_NONLINEAR` is missing?

Reply:
No. That's handled a few lines later

Reply:
Duhh... I see that now!